python/extensions.bzl

```diff
@@ -127,3 +133,88 @@ pip = module_extension(
+# that is not passed into the current context.
+# The module_label is the key element that is passed in.
+# This value provides the root location of the labels
+# See https://bazel.build/external/extension#repository_names_and_visibility
+def _repo_mapped_label(module_label, extension_name, apparent):
+    """Construct a canonical repo label accounting for repo mapping.
+
+    Args:
+        module_label: Label object of the module hosting the extension; see
+          "_module" implicit attribute.
+        extension_name: str, name of the extension that created the repo in `apparent`.
+        apparent: str, a repo-qualified target string, but without the "@". e.g.
+          "python38_x86_linux//:python". The repo name should use the apparent
+          name used by the extension named by `ext_name` (i.e. the value of the
+          `name` arg the extension passes to repository rules)
+    """
+    return Label("@@{module}~{extension_name}~{apparent}".format(
+        module = module_label.workspace_name,
+        extension_name = extension_name,
+        apparent = apparent,
+    ))
+
+# We are doing some bazel stuff here that could use an explanation.
+# The basis of this function is that we need to create a symlink to
+# the python binary that exists in a different repo that we know is
+# setup by rules_python.
+#
+# We are building a Label like
+# @@rules_python~override~python~python3_x86_64-unknown-linux-gnu//:python
+# and then the function creates a symlink named python to that Label.
+# The tricky part is the "~override~" part can't be known in advance
+# and will change depending on how and what version of rules_python
+# is used. To figure that part out, an implicit attribute is used to
+# resolve the module's current name (see "_module" attribute)
+#
+# We are building the Label name dynamically, and can do this even
+# though the Label is not passed into this function.  If we choose
+# not do this a user would have to write another 16 lines
+# of configuration code, but we are able to save them that work
+# because we know how rules_python works internally.  We are using
+# functions from private:toolchains_repo.bzl which is where the repo
+# is being built. The repo name differs between host OS and platforms
+# and the functions from toolchains_repo gives us this functions that
+# information.
+def _host_hub_impl(repo_ctx):
+    repo_ctx.file("BUILD.bazel", "")
+
+    # The two get_ functions we use are also utilized when building
+    # the repositories for the different interpreters.
+    (os, arch) = get_host_os_arch(repo_ctx)
+    host_platform = "{}_{}//:python".format(
+        repo_ctx.attr.extension_name,
+        get_host_platform(os, arch),
+    )
+
+    # the attribute is set to attr.label(default = "//:_"), which
+    # provides us the resolved, canonical, prefix for the module's repos.
+    # The extension_name "python" is created by the rules_python
+    # folder structure.  Ergo rules_python/python.
```

Comment:
This bit of the comment looks out of date? The repo_mapped_label function docs this pretty well, so remove.

```suggestion
```

Reply:
I'm trying to explain why we have "python" hardcoded.  It is because of the folder structure in rules_python. What do you recommend?

```
label = _repo_mapped_label(repo_ctx.attr._module, "python", host_platform)
```

We are hardcoding "python" in the line above.